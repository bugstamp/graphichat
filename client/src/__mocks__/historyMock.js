export const historyPushMock = jest.fn(str => str);

const historyMock = {
  push: historyPushMock,
};

export default historyMock;
