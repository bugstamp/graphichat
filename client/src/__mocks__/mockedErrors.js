import { createError } from 'apollo-errors';

export const AuthenticationError = createError('AuthenticationError', {
  message: 'The user is not authorized',
});

export const BadInputError = createError('BadInputError', {
  message: 'The provided value is invalid',
});

export const ForbiddenError = createError('ForbiddenError', {
  message: 'Access is denied',
});

export const ServerError = createError('ServerError', {
  message: 'Server Error',
});

export class GraphQLErrors extends Error {
  constructor(graphQLErrors) {
    super();
    this.graphQLErrors = graphQLErrors;
    this.name = 'GraphQLErrors';
  }
}
