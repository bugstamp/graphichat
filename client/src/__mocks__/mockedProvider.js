import React from 'react';
import { MockedProvider } from '@apollo/react-testing';
import { mount } from 'enzyme';

export const mountMockedProvider = (children, cache, mocks = []) => mount((
  <MockedProvider mocks={mocks} cache={cache}>
    {children}
  </MockedProvider>
));
