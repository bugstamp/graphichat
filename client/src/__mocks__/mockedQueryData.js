export const me = {
  id: 1,
  avatar: {
    sm: 'smAvatar',
    md: 'mdAvatar',
    __typename: 'UserAvatar',
  },
  username: 'me',
  displayName: 'Me Test',
  firstName: 'Me',
  lastName: 'Test',
  status: 'COMPLETED',
  lastDate: Date.now(),
  __typename: 'User',
};
