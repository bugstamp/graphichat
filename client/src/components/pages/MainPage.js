import React, {
  cloneElement,
  useState,
  useEffect,
} from 'react';
import PropTypes from 'prop-types';

import Grid from '@material-ui/core/Grid';
import Hidden from '@material-ui/core/Hidden';

import Navigation from '../common/Navigation/Navigation';
import SettingsDialog from './SettingsDialog';
import { AppContainer, MainPageWrapper } from './styled';

import { getAvatar } from '../../helpers';
import { meProps } from '../propTypes';

const MainPage = (props) => {
  const {
    children,
    loading,
    me,
    avatarUploading,
    uploadAvatar,
    updating,
    updateUser,
    userUpdatingError,
    signOut,
    sessionExpired,
  } = props;
  const [settingsDialog, toggleSettingsDialog] = useState(false);
  const handleToggleSettingDialog = () => {
    toggleSettingsDialog(!settingsDialog);
  };
  const handleSignOut = () => {
    const { id } = me;

    signOut({ variables: { userId: id } });
  };
  useEffect(() => {
    if (sessionExpired) {
      handleSignOut();
    }
  }, [sessionExpired]);
  const avatar = getAvatar(me, 'md');
  const childrenCloneProps = {
    initialLoading: loading,
    toggleSettingsDialog: handleToggleSettingDialog,
    signOut: handleSignOut,
  };
  const childrenClone = children
    ? cloneElement(children, childrenCloneProps)
    : null;

  return (
    <AppContainer container spacing={0} justify="center">
      <MainPageWrapper>
        <Grid container spacing={0}>
          <Hidden smDown>
            <Grid item>
              <Navigation
                toggleSettingsDialog={handleToggleSettingDialog}
                signOut={handleSignOut}
              />
            </Grid>
          </Hidden>
          <Grid item xs>
            {childrenClone}
          </Grid>
        </Grid>
      </MainPageWrapper>
      <SettingsDialog
        me={me}
        open={settingsDialog}
        toggle={handleToggleSettingDialog}
        avatar={avatar}
        avatarUploading={avatarUploading}
        uploadAvatar={uploadAvatar}
        updating={updating}
        updateUser={updateUser}
        error={userUpdatingError}
      />
    </AppContainer>
  );
};

MainPage.defaultProps = {
  children: null,
  me: {},
  userUpdatingError: null,
};
MainPage.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.node]),
  loading: PropTypes.bool.isRequired,
  me: PropTypes.shape(meProps),
  sessionExpired: PropTypes.bool.isRequired,
  avatarUploading: PropTypes.bool.isRequired,
  signOut: PropTypes.func.isRequired,
  uploadAvatar: PropTypes.func.isRequired,
  updating: PropTypes.bool.isRequired,
  updateUser: PropTypes.func.isRequired,
  userUpdatingError: PropTypes.instanceOf(Error),
};

export default MainPage;
