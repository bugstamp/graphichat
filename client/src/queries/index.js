import * as auth from './auth';
import * as user from './user';
import * as chat from './chat';

export default {
  auth,
  user,
  chat,
};
