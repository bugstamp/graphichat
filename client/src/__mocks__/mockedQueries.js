import gql from '../gql';

import { me } from './mockedQueryData';

const {
  GET_INITIAL_DATA,
  UPDATE_USER,
  USER_UPDATE_SUBSCRIPTION,
} = gql;

export const getInitialDataMock = {
  request: {
    query: GET_INITIAL_DATA,
    variables: {},
  },
  result: {
    data: {
      me,
      myContacts: [],
      myChats: [],
    },
  },
};
export const updateUserMock = {
  request: {
    query: UPDATE_USER,
    variables: {
      field: 'username',
      value: 'user',
    },
  },
  result: {
    data: {
      updateUser: me,
    },
  },
};
export const userUpdateSubscriptionMock = {
  request: {
    query: USER_UPDATE_SUBSCRIPTION,
    variables: {},
  },
  result: {
    data: {
      userUpdated: me,
    },
  },
};
