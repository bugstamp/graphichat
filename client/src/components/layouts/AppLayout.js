import React from 'react';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader';
import { get } from 'lodash';

import AppLayoutContainer from '../containers/AppLayoutContainer';
import MainPage from '../pages/MainPage';

const AppLayout = ({ children }) => (
  <AppLayoutContainer>
    {({
      getInitialData: {
        data: initialData,
        loading,
      },
      signOut: {
        mutation: signOut,
      },
      uploadAvatar: {
        mutation: uploadAvatar,
        result: {
          loading: avatarUploading,
        },
      },
      checkSessionExpiration: {
        data: sessionData,
      },
      updateUser: {
        mutation: updateUser,
        result: {
          loading: updating,
          error: userUpdatingError,
        },
      },
    }) => {
      const me = get(initialData, 'me', {});
      const sessionExpired = get(sessionData, 'sessionExpired', false);

      return (
        <MainPage
          loading={loading}
          me={me}
          signOut={signOut}
          avatarUploading={avatarUploading}
          uploadAvatar={uploadAvatar}
          updating={updating}
          updateUser={updateUser}
          userUpdatingError={userUpdatingError}
          sessionExpired={sessionExpired}
        >
          {children}
        </MainPage>
      );
    }}
  </AppLayoutContainer>
);

AppLayout.defaultProps = {
  children: null,
};
AppLayout.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.node]),
};

export default hot(module)(AppLayout);
