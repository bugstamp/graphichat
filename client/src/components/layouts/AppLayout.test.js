import React from 'react';
import wait from 'waait';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { shallow } from 'enzyme';
import { act } from 'react-dom/test-utils';

import {
  getInitialDataMock,
  updateUserMock,
  userUpdateSubscriptionMock,
} from '../../__mocks__/mockedQueries';
import { mountMockedProvider } from '../../__mocks__/mockedProvider';
import queries from '../../queries';

import AppLayout from './AppLayout';
import MainPage from '../pages/MainPage';

describe('test AppLayout', () => {
  test('snapshot render', () => {
    const wrapper = shallow(<AppLayout />);

    expect(wrapper).toMatchSnapshot();
  });
});
