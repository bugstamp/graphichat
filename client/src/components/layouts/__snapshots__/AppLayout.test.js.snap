// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AppLayout snapshot render 1`] = `
<Adopt(Unknown)(Unknown)(Unknown)(Unknown)(Unknown)(Unknown)(Unknown)(Unknown)>
  <Component />
</Adopt(Unknown)(Unknown)(Unknown)(Unknown)(Unknown)(Unknown)(Unknown)(Unknown)>
`;
